Assets/MRTK/Services/CameraSystem/ProjectionOverride.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using UnityEngine;
+using System.Runtime.InteropServices;
+using Microsoft.MixedReality.Toolkit.Utilities;
+
+#if WINDOWS_UWP
+using Windows.Perception.Spatial;
+using Windows.Graphics.Holographic;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.CameraSystem
+{
+    /// <summary>
+    /// Helper class to override the projection parameters of the HoloLens frame being presented
+    /// so that smaller details may appear more sharp.  The FOV of the HoloLens will be smaller
+    /// as a tradeoff.
+    /// </summary>
+    /// <remarks>
+    /// Instances of this class are created dynamically by <see cref="MixedRealityCameraSystem"/>.
+    /// So there is no need for an AddComponentMenu attribute.
+    /// </remarks>
+    internal class ProjectionOverride : MonoBehaviour
+    {
+        /// <summary>
+        /// When this is true, projection will be overridden on each frame
+        /// </summary>
+        public bool ReadingModeEnabled { get; set; } = false;
+
+#if WINDOWS_UWP
+        /// <summary>
+        /// Internal HoloLens Frame Structure.  Documented <see href="https://docs.microsoft.com/en-us/windows/mixed-reality/develop/unity/unity-xrdevice-advanced">here</see>.
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential)]
+        private struct HolographicFrameNativeData
+        {
+#pragma warning disable 0649
+            public uint VersionNumber;
+            public uint MaxNumberOfCameras;
+            public IntPtr ISpatialCoordinateSystemPtr;
+            public IntPtr IHolographicFramePtr;
+            public IntPtr IHolographicCameraPtr;
+#pragma warning restore 0649
+        }
+
+        /// <summary>
+        /// Coroutine function to set the camera matrices back to their defaults
+        /// </summary>
+        /// <returns>Coroutine enumerator</returns>
+        IEnumerator ResetViewMatricesOnFrameEnd()
+        {
+            yield return new WaitForEndOfFrame();
+            Camera cam = GetComponent<Camera>();
+            cam.ResetStereoViewMatrices();
+            cam.ResetStereoProjectionMatrices();
+        }
+
+        /// <inheritdoc />
+        void OnPreCull()
+        {
+            if (!ReadingModeEnabled)
+            {
+                return;
+            }
+
+            StartCoroutine(ResetViewMatricesOnFrameEnd());
+
+            IntPtr nativeStruct = UnityEngine.XR.XRDevice.GetNativePtr();
```

Comment:
A quick note that this API unfortunately is no longer valid with the new XR APIs. The usage here is also specific to WMR, which feels weird to call in a platform-agnostic service like this.